### ISSUE TITLE
Definition of Datasets

### ISSUE THREAD

**Issue by EiffL** [deepskies/dsutils#30]
Probably the first thing that needs to be figured out, and even independently from the rest, is how to expose datasets. Here is a wishlist:
 -  [ ] Should be accessible in one line:
```
import dsutils as ds
data = ds.get_dataset('strong_lens')
```
  - [ ] Should contain reference to paper introducing the dataset 

  - [ ] Should be TF and PyTorch friendly

  - [ ] Probably should have a separate community managed repo for astro datasets implementing the API defined here

Food for thoughts:
  - Tensorflow Datasets:  https://www.tensorflow.org/datasets
  - Torchvision Datasets: https://pytorch.org/docs/stable/torchvision/datasets.html

**Comment by EiffL** [comment]
I'm going to tag @tomcharnock here as I think he might have some thoughts on it ;-)

**Comment by tomcharnock** [comment]
Yep, happy to help. I’ve just been wrestling with tf.data.Datasets all day and I think it would be totally possible to make an easily callable dataset based on it. I can’t vouch for pytorch (which keeps autocorrecting to pyrotechnics) though.
